### PR TITLE
Challenge51: Misconfiguration of docker secret in code.

### PR DESCRIPTION
**Pull request by Shubham-Patel07**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description
New challenge created for misconfiguration of docker secret in code.

### Relations

Relates #811 

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [x] I tested the changes in this PR (if applicable)
-   [x] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [ ] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

File: src/main/resources/wrong-secrets-configuration.yaml [hunk 1]
```diff
@@ -775,3 +775,16 @@ configurations:
...
+      sources:
+        - class-name: "org.owasp.wrongsecrets.challenges.docker.Challenge49"
+          explanation: "explanations/challenge49.adoc"
+          hint: "explanations/challenge49_hint.adoc"
+          reason: "explanations/challenge49_reason.adoc"
+          environments: *all_envs
+      difficulty: *easy
+      category: *secrets
+      ctf:
+        enabled: false
```
Comment (commjoen) [inline src/main/resources/wrong-secrets-configuration.yaml:790]:
```suggestion
        enabled: true
```

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -72,6 +72,7 @@ challenge26ciphertext=gbU5thfgy8nwzF/qc1Pq59PrJzLB+bfAdTOrx969JZx1CKeG4Sq7v1uUpz
 DEFAULT37=DEFAULT37
 challenge27ciphertext=gYPQPfb0TUgWK630tHCWGwwME6IWtPWA51eU0Qpb9H7/lMlZPdLGZWmYE83YmEDmaEvFr2hX
 challenge41password=UEBzc3dvcmQxMjM=
+DOCKER_SECRET_CHALLENGE49=sdfjklhjUYYkghjv;-jhkf
```
Comment (commjoen) [inline src/main/resources/application.properties:75]: Can we maybe encode/encrypt this in order to give it away? we could maybe work together on having a shell script (see https://github.com/OWASP/wrongsecrets/blob/master/.github/scripts/docker-create.sh) to generate the secret in [the](https://github.com/OWASP/wrongsecrets/pull/1611/files#diff-d9c0c07f029efb8a4de5742d0b584dde1cc286a6b76cac7daf24bf781ac89f27) and here. But I am fine with doing this post PR as well :) .

Comment (Shubham-Patel07) [reply src/main/resources/application.properties:75]: You mean generate different secret everytime ??

Comment (commjoen) [reply src/main/resources/application.properties:75]: Yes, or just encode the secret here. What would be your take on this @bendehaan ?

Comment (bendehaan) [reply src/main/resources/application.properties:75]: I would prefer randomization on startup - less noise on current scanners, and lower likelihood contestants in a CTF just find the value in a walkthrough on the web

Comment (commjoen) [reply src/main/resources/application.properties:75]: Lets skip the randomization for now :-)

**Pull request closed without merging**